x/ibc/testing/chain.go

```diff
@@ -149,7 +161,10 @@ func (chain *TestChain) QueryProof(key []byte) ([]byte, uint64) {
 	proof, err := chain.App.AppCodec().MarshalBinaryBare(&merkleProof)
 	require.NoError(chain.t, err)
 
-	return proof, uint64(res.Height)
+	// proof height + 1 is returned as the proof created corresponds to the height the proof
+	// was created in the IAVL tree. Tendermint and subsequently the clients that rely on it
+	// have heights 1 above the IAVL tree. Thus we return proof height + 1
+	return proof, uint64(res.Height) + 1
```

Comment:
Note for reviewers @cwgoes @fedekunze : This is an issue that me and @jackzampolin repeatedly ran into when working with the relayer. We fixed the problem by incrementing the query height before passing into proof height.

We should consider whether we want to make this a requirement for relayers to handle correctly or if we want to deal with this peculiarity on the SDK side.

Reply:
I think we should open up an issue to discuss this

Reply:
yeah let's create an issue 👍